src/main/java/edu/njit/jerse/ashe/utils/JavaCodeParser.java

```diff
@@ -98,34 +101,42 @@ public record MethodSignature(
     /**
      * Extracts method signature from a given method string.
      *
-     * @param method the method string from which to extract the signature.
-     * @return an Optional containing {@link MethodSignature} if found, else empty.
+     * @param method the method string from which to extract the signature
+     * @return {@link MethodSignature} representing the extracted method signature
```

Comment:
Add `@throws` clause to the doc indicating that a `NoSuchElementException` is thrown if extraction fails. (Also, add `throws NoSuchElementException` to the method signature.)